it/src/main/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManager.java

```diff
@@ -0,0 +1,421 @@
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminSettings;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateInstanceRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link BigtableResourceManager} interface.
+ *
+ * <p>The class supports one instance, and multiple tables per manager object. An instance is
+ * created when the first table is created if one has not been created already.
+ *
+ * <p>The instance id is formed using testId. The instance id will be "{testId}-{ISO8601 time,
+ * microsecond precision}", with additional formatting. Note: If testId is more than 30 characters,
+ * a new testId will be formed for naming: {first 21 chars of long testId} + “-” + {8 char hash of
+ * testId}.
+ *
+ * <p>The class is thread-safe.
+ */
+public class DefaultBigtableResourceManager implements BigtableResourceManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultBigtableResourceManager.class);
+  private static final int MAX_TEST_ID_LENGTH = 30;
+  private static final String DEFAULT_CLUSTER_ZONE = "us-central1-a";
+  private static final int DEFAULT_CLUSTER_NUM_NODES = 1;
+  private static final StorageType DEFAULT_CLUSTER_STORAGE_TYPE = StorageType.SSD;
+
+  private final String projectId;
+  private final String instanceId;
+  private final String testId;
+  private final BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler;
+
+  private boolean hasInstance = false;
+
+  @VisibleForTesting
+  DefaultBigtableResourceManager(
+      String testId, String projectId, CredentialsProvider credentialsProvider) throws IOException {
+    this(
+        DefaultBigtableResourceManager.builder(testId, projectId)
+            .setCredentialsProvider(credentialsProvider));
+  }
+
+  @VisibleForTesting
+  DefaultBigtableResourceManager(
+      String testId,
+      String projectId,
+      BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler)
+      throws IOException {
+    this(
+        bigtableResourceManagerClientHandler,
+        DefaultBigtableResourceManager.builder(testId, projectId));
+  }
+
+  private DefaultBigtableResourceManager(DefaultBigtableResourceManager.Builder builder)
+      throws IOException {
```

Comment:
I suggest refactoring all these constructors:
1) Keep only this one (that accepts a Builder).
2) Builder should keep an instance of `BigtableResourceManagerClientHandler`, not 3 individual `Bigtable*Settings` objects.
3) Tests can do `Builder#setBigtableResourceManagerClientHandler()` to override the handler with a mock implementation.
4) If no overrides provided, Builder will create the default handler instance in the `build()` method.

This way we'll have only 1 constructor instead of 4, no need for `@VisibleForTesting` annotations, and no constructors like `DefaultBigtableResourceManager(BigtableResourceManagerClientHandler, Builder)`, which actually accepts 2 sets of bigtable settings at the same time (one in the handler argument and one in the builder).

Reply:
This was another instance of trying to reflect the pattern used in the Spanner RM, however, I agree it is clunky. I refactored as you suggested, and it looks much cleaner.